apps/keyboard/js/views/candidate_panel_view.js

```diff
@@ -0,0 +1,249 @@
+};
+
+CandidateToggleButtonView.prototype.hide = function() {
+  this.element.classList.add('hide');
+};
+
+/**
+ * CandidatePanelView handles the rendering of candidate panel, used for showing
+ * word suggestions.
+ */
+function CandidatePanelView(target, options, viewManager) {
+  this.element = null;
+  this.className = options.className;
+  this.viewManager = viewManager;
+  this.widthUnit = options.widthUnit;
+}
+
+// Default value for how many candidates shown in a row.
+CandidatePanelView.prototype.countPerRow = 8;
+
+CandidatePanelView.prototype.render = function render() {
+  var candidatePanel = document.createElement('div');
+  candidatePanel.setAttribute('role', 'group');
+  candidatePanel.dataset.l10nId = 'wordSuggestions2';
+
+  candidatePanel.classList.add('keyboard-candidate-panel');
+  if (this.className) {
+    candidatePanel.classList.add(this.className);
+  }
+
+  var suggestionsContainer = document.createElement('div');
+  suggestionsContainer.classList.add('suggestions-container');
+  suggestionsContainer.setAttribute('role', 'listbox');
+  candidatePanel.appendChild(suggestionsContainer);
+  this.suggestionsContainer = suggestionsContainer;
+
+  // Create toggle button
+  // we're not getting reference of LayoutManager, so define this manually
+  var KEYCODE_TOGGLE_CANDIDATE_PANEL = -4;
+
+  this.toggleButton = new CandidateToggleButtonView({
+    keyCode: KEYCODE_TOGGLE_CANDIDATE_PANEL },
+    null,
+    this.viewManager);
+  this.toggleButton.render();
+
+  candidatePanel.appendChild(this.toggleButton.element);
+  this.element = candidatePanel;
+};
+
+//
+// Show candidates
+// Each candidate is a string or an array of two strings
+//
+CandidatePanelView.prototype.showCandidates = function(candidates) {
+  if (!this.element) {
+    console.error('CandidatePanelView: CandidatePanel is not ready yet!');
+    return;
+  }
+
+  var candidatePanel = this.element;
+  candidatePanel.dataset.candidateIndicator = 0;
+  this.suggestionsContainer.innerHTML = '';
+  this.toggleButton.hide();
+
+  var docFragment = this.renderCandidatePanel(1, candidates, true);
+
+  this.suggestionsContainer.appendChild(docFragment);
+};
+
+CandidatePanelView.prototype.renderCandidatePanel = function(rowLimit,
+                                                            candidates,
+                                                            indentFirstRow) {
+  var candidatePanel = this.element;
+
+  var docFragment = document.createDocumentFragment();
+  if (candidates.length === 0) {
+    candidatePanel.dataset.rowCount = 0;
+    return docFragment;
+  }
+
+  var rowDiv = document.createElement('div');
+  rowDiv.classList.add('candidate-row');
+  if (indentFirstRow) {
+    rowDiv.classList.add('candidate-row-first');
+  }
+
+  var nowUnit = 0;
+  var rowCount = 0;
+
+  if (rowLimit < 0) {
+    rowLimit = Number.Infinity;
+  }
+
+  var candidatesLength = candidates.length;
+
+  for (var i = 0; i < candidatesLength; i++) {
```

Comment:
You could use `Array#some` to loop the element and `return true` to "break" the loop.

http://mdn.io/array%20some